sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/DiamondOperatorForVariableDefinitionCheck.java

```diff
@@ -113,4 +113,46 @@ private static DetailAST getFirstTypeArgumentsToken(DetailAST rootToken) {
         return resultNode;
     }
 
+    /**
+     * Checks if the 2 given trees have the same children, type, and text.
+     * @param left One of the trees to compare.
+     * @param right The other tree to compare.
+     * @return {@code true} if the trees are equal.
+     */
+    private static boolean isTreeEqual(DetailAST left, DetailAST right) {
+        boolean result;
+
+        if (isAstEqual(left, right)) {
+            result = true;
+
+            DetailAST leftChild = left.getFirstChild();
+            DetailAST rightChild = right.getFirstChild();
+
+            while (leftChild != rightChild) {
+                if (!isTreeEqual(leftChild, rightChild)) {
```

Comment:
There is no protection for one being `null` and the other not being `null`. Regression couldn't find any cases.
We check `type` when scrolling through the children, so for one to be `null` we would need a token that has an optional child. I couldn't come up with any cases.